py/desispec/tsnr.py

```diff
@@ -455,6 +457,71 @@ def fb_rdnoise(fibers, frame, psf):
 
     return rdnoise
 
+def var_tracer(tracer, frame, angperspecbin, fiberflat, fluxcalib, exposure_seeing_fwhm=1.1):
+    '''
+    Source Poisson term to the model ivar, following conventions defined at:
+        https://desi.lbl.gov/trac/wiki/SurveyOps/SurveySpeed.  
+
+    See also:
+        https://github.com/desihub/desispec/blob/master/py/desispec/efftime.py
+
+    Args:
+        tracer: [bgs, backup] string, defines program.
+        frame: desispec.frame instance
+        angperspecbin: float, angstroms per bin in spectral reductions
+        fiberflat: desispec instance
+        fluxcalib: desispec instance
+        fiber_diameter_arcsec:
+        
+    Returns: 
+        nominal flux [e/specbin] corresponding to frame.wave
+    '''
+    log = get_logger()
+
+    fiber_params = load_desiparams()['fibers']
+    fiber_dia = fiber_params['diameter_um'] # 107 um.
+    fiber_dia_asec = fiber_params['diameter_arcsec'] # 1.52 ''
+    
+    fiber_area_arcsec2 = np.pi*(fiber_dia_asec/2.)**2
+    
+    if tracer == 'bgs':
+        # Nominal fiberloss dependence on seeing.  Assumes zero offset. 
+        fiberfrac = np.exp(0.0341 * np.log(exposure_seeing_fwhm)**3 -0.3611 * np.log(exposure_seeing_fwhm)**2 -0.7175 * np.log(exposure_seeing_fwhm) -1.5643)
+
+        # Note: neglects transparency & EBV corrections. 
+        nominal = 15.8                    # r=19.5 [nanomaggie].    
+
+    elif tracer == 'backup':
+        fiberfrac = np.exp(0.0989 * np.log(exposure_seeing_fwhm)**3 -0.5588 * np.log(exposure_seeing_fwhm)**2 -0.9708 * np.log(exposure_seeing_fwhm) -0.4473)
+        nominal = 27.5                    # r=18.9 [nanomaggie].
+        
+    else:
+        # No source poisson term otherwise. 
+        nominal = 0.0                     # [nanomaggie]. 
+        fiberfrac = 1.0
+
+        return nominal                    # [e/specbin].
+
+    
+    nominal *= fiberfrac
+    nominal /= fiber_area_arcsec2         # [nMg / ''].
```

Comment:
I don't think we should divide by the fiber area here because the output should be in `(electrons/specbin)` (it's done differently on the wiki because there it is to be added to the sky flux per arcsec2).